### PR TITLE
fix(time-series-card): fix refs in mocks for test output

### LINKED ISSUE #2453: [TimeSeriesCard] Function components cannot be given refs
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

While cleaning up testing output we've uncovered some other bugs that may introduce breaking changes that need to be fixed in v3.
```
  ● Console

    console.error
      Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
      
      Check the render method of `TimeSeriesCard`.
          in Unknown (created by TimeSeriesCard)
          in div (created by TimeSeriesCard)
          in div (created by CardContent)
          in CardContent (created by SizeMe)
          in div (created by CardWrapper)
          in CardWrapper (created by SizeMe)
          in Unknown (created by SizeMeRenderer(Component))
          in SizeMeReferenceWrapper (created by SizeMeRenderer(Component))
          in SizeMeRenderer(Component) (created by SizeMe(Component))
          in SizeMe(Component) (created by SizeMe)
          in SizeMe (created by VisibilitySensor)
          in VisibilitySensor (created by Card)
          in Card (created by TimeSeriesCard)
          in TimeSeriesCard (created by WrapperComponent)
          in WrapperComponent

      at printWarning (../../node_modules/react-dom/cjs/react-dom.development.js:88:30)
      at error (../../node_modules/react-dom/cjs/react-dom.development.js:60:5)
      at validateFunctionComponentInDev (../../node_modules/react-dom/cjs/react-dom.development.js:17578:9)
      at mountIndeterminateComponent (../../node_modules/react-dom/cjs/react-dom.development.js:17545:7)
      at beginWork (../../node_modules/react-dom/cjs/react-dom.development.js:18596:16)
      at beginWork$1 (../../node_modules/react-dom/cjs/react-dom.development.js:23179:14)
      at performUnitOfWork (../../node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at workLoopSync (../../node_modules/react-dom/cjs/react-dom.development.js:22130:22)
```

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #2453 

**Summary**

- Testing output was throwing a "Functional components can not be given refs" error when using TimeSeriesCard. This PR fixes it by wrapping the mocked charts with React.forwardRef. This ended up not being a v3 change, so basing it against next.

**Change List (commits, features, bugs, etc)**

- add React.forwardRef calls to all mocked carbon charts
- fix useRef references in TimeSeriesCard

**Acceptance Test (how to verify the PR)**

- No testing noise when testing TimeSeriesCard (except for testID prop deprecation notices)

**Regression Test (how to make sure this PR doesn't break old functionality)**

- all TimeSeriesCard tests still pass and [stories](https://deploy-preview-2678--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timeseriescard--multi-line-interval-data-choices) function as expected.

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/TimeSeriesCard/TimeSeriesCard.jsx [hunk 1]
```diff
@@ -553,7 +553,7 @@ const TimeSeriesCard = ({
           >
             <ChartComponent
               ref={(el) => {
-                chartRef = el;
+                chartRef.current = el;
```
Comment (davidicus) [inline packages/react/src/components/TimeSeriesCard/TimeSeriesCard.jsx:556]: why do we need to pass in the ref like this instead of passing it as prop?

**Review by davidicus — APPROVED** [review]
LGTM. One question about the ref

File: packages/react/src/components/TimeSeriesCard/TimeSeriesCard.jsx (hunk 1 above)
Comment (kevinsperrine) [reply packages/react/src/components/TimeSeriesCard/TimeSeriesCard.jsx:556]: That's fair. That's just the way it was, and I did't even think to change it.

**Review by bjornalm — APPROVED** [review]
Nice work

**Pull request merged**